src/pyscaffoldext/custom_extension/custom_extension.py

```diff
@@ -14,18 +14,27 @@
 EXTENSION_FILE_NAME = "extension"
 
 
+class InvalidProjectNameException(BaseException):
```

Comment:
Nice, it's good to have an own exception for that. Following the module `exceptions.py` in PyScaffold you could also do:
```
class InvalidProjectNameException(RuntimeError):
    """Project name does not comply with convention of an extension"""

    DEFAULT_MESSAGE = (
        "An extension project name should start with "
        "``pyscaffoldext-`` or use ``--force`` to overwrite.")

    def __init__(self, message=DEFAULT_MESSAGE, *args, **kwargs):
        super().__init__(message, *args, **kwargs)
```
With this you would save later the specification of the actual error message.